File: core/src/main/java/com/emarsys/core/concurrency/CoreSdkHandler.java
package com.emarsys.core.concurrency;

import android.os.Handler;
import android.os.HandlerThread;
import android.os.Message;

import com.emarsys.core.util.log.Logger;
import com.emarsys.core.util.log.entry.CrashLog;

public class CoreSdkHandler extends Handler {

    public CoreSdkHandler(HandlerThread handlerThread) {
        super(handlerThread.getLooper());
    }

    @Override
    public void dispatchMessage(Message msg) {
        try {
            super.dispatchMessage(msg);
        } catch (Exception e) {
            Logger.error(new CrashLog(e));
        }
    }
}

File: core/src/main/java/com/emarsys/core/handler/CoreSdkHandler.kt
package com.emarsys.core.handler;

import android.os.Handler
import com.emarsys.core.Mockable

@Mockable
class CoreSdkHandler(final val handler: Handler) {
    fun post(runnable: Runnable) {
        handler.post(runnable)
    }

    val looper = handler.looper
}